chrome-extension/content.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Content script for all urls 
+ */
+
+//Insert bulma stylesheet
+//TODO: STOP USING BULMA ONCE I WRITE MY OWN CSS FOR ALL EXTENSION ELEMENTS
+const bulmaLink = document.createElement('link');
+bulmaLink.setAttribute('href','https://cdn.jsdelivr.net/npm/bulma@0.9.0/css/bulma.min.css');
+bulmaLink.setAttribute('rel','stylesheet');
+document.head.appendChild(bulmaLink)
+
+//Inject invisible sidebar onto page
+const sidebar = document.createElement('div');
+sidebar.id = 'lobstericecream';
+sidebar.classList.add('panel','injection-panel','px-4','py-4');
+document.body.appendChild(sidebar); 
+
+//Inject invisible highlight icon onto page
+const iconButton = document.createElement('button');
+iconButton.classList.add('injection-icon');
+iconButton.innerHTML = 'Save';
+//TODO: CHANGE BUTTON TO AN ACTUAL ICON!! 
+
+document.body.appendChild(iconButton);
+
+//Keep track of last used highlight
+let lastContent; 
+
+//Show extension icon when selecting text on page
+document.onclick = (event) => {
```

Comment:
I'm not well-versed in JS so pardon if this doesn't make sense, but it seems like onselect might be a useful event listener to use in this script. I think it could make some of the semantics more clear at the very least. That being said, I think we may want to consider transitioning our functionality to the context menu. Either that, or we could add the ability to turn the extension on and off. As it currently is, I find the extension a bit intrusive when I'm not trip clipping.

Reply:
Discussed offline :)